dask/array/core.py

```diff
@@ -1176,7 +1178,7 @@ def from_func(func, shape, dtype=None, name=None, args=(), kwargs={}):
     """
     if args or kwargs:
         func = partial(func, *args, **kwargs)
-    name = name or next(names)
+    name = name or 'from_func-' + tokenize((func, shape, dtype, args, kwargs))
```

Comment:
I suggest that we pass many arguments to `tokenize` and we normalize each of them but don't recurse further than that.  This is done in the tiny example I have above.